File: Programmers/Lv2/전화번호 목록.js
function solution(phone_book) {
  var answer = true;
  phone_book.sort();

  for (let i = 0; i < phone_book.length - 1; i++) {
    var k = phone_book[i].length;
    if (phone_book[i] === phone_book[i + 1].substring(0, k)) {
      return false;
    }
  }
  return answer;
}

File: Programmers/Lv2/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    sort(phone_book.begin(), phone_book.end());
    
    for(int i = 0; i < phone_book.size() - 1; i++){
        int k = phone_book[i].size();
        string tmp = phone_book[i + 1].substr(0, k);
        if(phone_book[i] == tmp){
            return false;
        }
    }
    return true;
}